tutorials/Inference-Bert-Model-for-High-Performance-with-ONNX-Runtime-on-AzureML.ipynb

```diff
@@ -0,0 +1,690 @@
+    "print(\"ONNX Runtime inference time: \", end - start)\n",
+    "\n",
+    "print(\"onnx runtime res \", res)\n",
+    "print()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3 - Deploy Bert model with ONNX Runtime through AzureML\n",
+    "\n",
+    "In Step 1 and 2, we have prepared an optimized ONNX Bert model and now we can deploy this model as a web service using Azure Machine Learning services and the ONNX Runtime.\n",
+    "\n",
+    "We're now going to deploy our ONNX model on Azure ML using the following steps.\n",
+    "\n",
+    "1. **Register our model** in our Azure Machine Learning workspace\n",
+    "2. **Write a scoring file** to evaluate our model with ONNX Runtime\n",
+    "3. **Write environment file** for our Docker container image.\n",
+    "4. **Create and register a container image** for our model\n",
+    "5. **Deploy to the cloud** using an Azure Container Instances VM with CPU and use it to make predictions using ONNX Runtime Python APIs\n",
+    "6. **Classify sample text input** so we can explore inference with our deployed service.\n",
+    "\n",
+    "![End-to-end pipeline with ONNX Runtime](https://raw.githubusercontent.com/vinitra/models/gtc-demo/gtc-demo/E2EPicture.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3.0 - Check your AzureML environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check core SDK version number\n",
+    "import azureml.core\n",
+    "from PIL import Image, ImageDraw, ImageFont\n",
+    "import json\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "print(\"SDK version:\", azureml.core.VERSION)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load your Azure ML workspace\n",
+    "\n",
+    "We begin by instantiating a workspace object from the existing workspace created earlier in the configuration notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core import Workspace\n",
+    "\n",
+    "ws = Workspace.from_config()\n",
+    "print(ws.name, ws.location, ws.resource_group, sep = '\\n')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3.1 - Register your model with Azure ML\n",
+    "\n",
+    "Now we upload the model and register it in the workspace. In the following tutorial. we use the bert SQuAD model outputted from Step 1 and 2 as an example. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.model import Model\n",
+    "\n",
+    "model = Model.register(model_path = \"./bert_azureml_large_uncased.onnx\",\n",
```

Comment:
model registration should always specify framework type

Reply:
framework and framework version. 

see the onnx deploy notebook here: https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/deployment/onnx/onnx-model-register-and-deploy.ipynb

model = Model.register(workspace=ws,
                       model_name='mnist-sample',                  # Name of the registered model in your workspace.
                       model_path='mnist-model.onnx',              # Local ONNX model to upload and register as a model.
                       model_framework=Model.Framework.ONNX ,      # Framework used to create the model.
                       model_framework_version='1.3',              # Version of ONNX used to create the model.
                       description='Onnx MNIST model')